pandas/tests/indexes/timedeltas/test_setops.py

```diff
@@ -74,3 +74,66 @@ def test_intersection_bug_1708(self):
         result = index_1 & index_2
         expected = timedelta_range('1 day 01:00:00', periods=3, freq='h')
         tm.assert_index_equal(result, expected)
+
+
+def test_intersection_bug_17391():
+    idx1 = pd.to_timedelta(range(3), unit='s')
+    idx2 = pd.to_timedelta(range(2, -1, -1), unit='s')
+    print(idx1)
+    print(idx2)
+    assert len(idx1.intersection(idx2)) == 3
```

Comment:
don't compare like this. construct the expected index as ordering is important here. use ``tm.assert_index_equal``